src/formatDate.js

```diff
@@ -1,14 +1,36 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        receivedDate['YY'] = dateArr[i].slice(2);
+        break;
+      case 'YY':
+        receivedDate['YYYY'] = (dateArr[i] <= 20 ? 20 : 19) + dateArr[i];
+        break;
+      default:
+        break;
+    }
+    receivedDate[fromFormat[i]] = dateArr[i];
```

Comment:
This line is overwriting the 'YYYY' or 'YY' value set in the switch statement above, which could lead to incorrect behavior. You should move this line into the `default` case of the switch statement to correctly assign 'MM' and 'DD' values.